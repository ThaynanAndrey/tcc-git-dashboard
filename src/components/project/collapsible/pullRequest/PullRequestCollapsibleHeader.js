import React from 'react';

const styleCollapsibleHeader = {
    display: "flex",
    justifyContent: "space-between"
};

const styleCollapsibleHeaderText = {
    display: "flex"
};

/**
 * Stateless component with the function of presenting a Collapsible Header to Pull Requests.
 * 
 * @author Thaynan Nunes
 */
const PullRequestCollapsibleHeader = ({ redirectPage, idProject }) => (
    <div className="collapsible-header" style={styleCollapsibleHeader}>
        <div style={styleCollapsibleHeaderText}>
            <i className="material-icons">dehaze</i>
            Pull Requests
        </div>
        <div>
            <button className="btn-small waves-effect waves-light green darken-2"
                    type="button" name="cadastrar-pr"
                    onClick={() => redirectPage(`${idProject}/adicionarPullRequests`)}>
                Adicionar
                <i className="material-icons left">add</i>
            </button>
        </div>
    </div>
);

export default PullRequestCollapsibleHeader;