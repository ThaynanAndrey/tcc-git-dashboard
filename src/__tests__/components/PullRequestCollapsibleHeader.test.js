import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import PullRequestCollapsibleHeader from '../../components/project/collapsible/pullRequest/PullRequestCollapsibleHeader';

describe('Test RepositoriesNoProject', () => {
  it('renders correctly a component to show PullRequestCollapsibleHeader', () => {
    const redirectPage = jest.fn();
    const idProject = '123';

    const tree = shallow(
      <PullRequestCollapsibleHeader 
        idProject={idProject}
        redirectPage={redirectPage}
      />
    );

    expect(toJson(tree)).toMatchSnapshot();
  });

  
  it('the component should have class collapsible-header', () => {
    const tree = shallow(
      <PullRequestCollapsibleHeader />
    );

    expect(tree.find(".collapsible-header").exists()).toBeTruthy();
  });
})