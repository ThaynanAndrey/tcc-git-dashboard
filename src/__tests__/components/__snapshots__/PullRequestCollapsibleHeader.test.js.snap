// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RepositoriesNoProject renders correctly a component to show PullRequestCollapsibleHeader 1`] = `
<div
  className="collapsible-header"
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-between",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <i
      className="material-icons"
    >
      dehaze
    </i>
    Pull Requests
  </div>
  <div>
    <button
      className="btn-small waves-effect waves-light green darken-2"
      name="cadastrar-pr"
      onClick={[Function]}
      type="button"
    >
      Adicionar
      <i
        className="material-icons left"
      >
        add
      </i>
    </button>
  </div>
</div>
`;
